deepmd/pt/model/model/spin_model.py

```diff
@@ -0,0 +1,459 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+from typing import (
+    Dict,
+    Optional,
+)
+
+import torch
+
+from deepmd.pt.utils.utils import (
+    dict_to_device,
+    to_torch_tensor,
+)
+from deepmd.utils.path import (
+    DPPath,
+)
+from deepmd.utils.spin import (
+    Spin,
+)
+
+from .dp_model import (
+    DPModel,
+)
+
+
+class SpinModel(torch.nn.Module):
```

Comment:
This class does not implement `format_nlist`, which was implemented in the atomic model. I noticed it is required by the C++ interface (see #3337). I am wondering whether we should copy/move it to `BaseModel` or add an abstract method to `BaseModel`.